File: server/src/models/message-model.js
import schema from "../lib/schema/message-schema.js";

/**
 * @param {String} id
 * @param {String} emailSender 
 * @param {String} emailReceiver 
 * @param {String} message 
 * @param {String} type 
 * @param {String} status 
 * @param {String} time 
 */
const insert = async({ id, emailSender, emailReceiver, message, type, status, time })=> {
    return await schema.create({ id, emailSender, emailReceiver, message, type, status, time });
}

/**
 * 
 * @param {String} email 
 */
const findSentMessages = async(email)=> {
    return await schema.find({emailReceiver: email, status: "sent"});
}

/**
 * 
 * @param {String} id 
 */
const deleteMessage = async(id)=> {
    return await schema.deleteOne({id});
}

export default {
    insert,
    findSentMessages,
    deleteMessage
};

File: app/lib/data/models/message_model.dart
class MessageModel {
  final String? id;
  final String emailSender;
  final String emailReceiver;
  final String message;
  final String type;
  final String status;
  final String time;

  MessageModel({
    this.id,
    required this.emailSender,
    required this.emailReceiver,
    required this.message,
    required this.type,
    required this.status,
    required this.time,
  });

  factory MessageModel.fromJson(Map<String, dynamic> json) {
    return MessageModel(
      id: json['id'] as String,
      emailSender: json['emailSender'] as String,
      emailReceiver: json['emailReceiver'] as String,
      message: json['message'] as String,
      type: json['type'] as String,
      status: json['status'] as String,
      time: json['time'] as String,
    );
  }

  Map<String, dynamic> toJson() {
    return <String, String>{
      'id': id!,
      'emailSender': emailSender,
      'emailReceiver': emailReceiver,
      'message': message,
      'type': type,
      'status': status,
      'time': time,
    };
  }
}
